dlogproofs/schnorr.go

```diff
@@ -63,7 +63,7 @@ func (prover *SchnorrProver) GetProofData(challenge *big.Int) (*big.Int, *big.In
 	z.Mod(z, prover.DLog.GetOrderOfSubgroup())
 
 	if prover.protocolType != common.ZKPOK {
-		return z, nil
+		return z, new(big.Int)
```

Comment:
This shouldn't be part of this refactoring PR. Even if it works wrong the way it is now, such a change deserves to be reviewed in a separate PR so that you can provide short explanation in its own commit message. Please revert this from this commit and open a new PR. Merging of the second PR should be trivial as there will be no conflicts.